pkgs/development/libraries/qt-6/patches/qtwebengine-xkb-includes.patch

```diff
@@ -0,0 +1,12 @@
+--- a/src/3rdparty/chromium/ui/events/ozone/layout/xkb/xkb_keyboard_layout_engine.cc	2023-04-19 21:58:29.127258300 +0900
++++ b/src/3rdparty/chromium/ui/events/ozone/layout/xkb/xkb_keyboard_layout_engine.cc	2023-04-19 22:01:35.860196284 +0900
+@@ -637,8 +637,7 @@
+                           .variant = layout_variant.c_str(),
+                           .options = ""};
+   std::unique_ptr<xkb_context, XkbContextDeleter> context;
+-  context.reset(xkb_context_new(XKB_CONTEXT_NO_DEFAULT_INCLUDES));
+-  xkb_context_include_path_append(context.get(), "/usr/share/X11/xkb");
++  context.reset(xkb_context_new(XKB_CONTEXT_NO_FLAGS));
```

Comment:
So how are the relevant files (set in the NixOS module) located with this patch applied? According to https://xkbcommon.org/doc/current/group__context.html, the context is created with an empty search path.

Reply:
And would these also apply to `chromium`, I'm not seeing special handling with xkb in our `chromium` packaging.

Reply:
My reading is that setting `XKB_CONTEXT_NO_DEFAULT_INCLUDES` (as it did before this patch) means it ignores `XKB_CONFIG_ROOT` (what NixOS sets).

After the patch, we only set `XKB_CONTEXT_NO_FLAGS`, which I believe means we get the default includes.

> And would these also apply to chromium

Nope, as @apfelkuchen6 pointed out (https://github.com/NixOS/nixpkgs/issues/226484#issuecomment-1510498373), this is in `#if defined(TOOLKIT_QT)`, so chromium doesn't build this code.

Reply:
Looked through the default include paths at https://xkbcommon.org/doc/current/group__include-path.html, sounds legit.